lib/plugins/aws/provider.js

```diff
@@ -1700,4 +1738,220 @@ class AwsProvider {
   }
 }
 
+Object.defineProperties(
+  AwsProvider.prototype,
+  memoizeeMethods({
+    getAccountId: d(
+      async function () {
+        const result = await this.getAccountInfo();
+        return result.accountId;
+      },
+      { promise: true }
+    ),
+    ensureDockerIsAvailable: d(
+      async function () {
+        try {
+          await spawnExt('docker', ['--version']);
+        } catch (err) {
+          throw new this.serverless.classes.Error(
+            'Could not find Docker installation. Ensure Docker is installed before continuing.'
+          );
+        }
+      },
+      { promise: true }
+    ),
+    dockerLoginToEcr: d(
+      async function () {
+        const registryId = await this.getAccountId();
+        const result = await this.request('ECR', 'getAuthorizationToken', {
+          registryIds: [registryId],
+        });
+        const { authorizationToken, proxyEndpoint } = result.authorizationData[0];
+        const decodedAuthToken = Buffer.from(authorizationToken, 'base64').toString().split(':')[1];
+        const dockerArgs = ['login', '--username', 'AWS', '--password-stdin', proxyEndpoint];
```

Comment:
I think docker has an option to simply pass password as an argument (`--password` ?).

It'll probably be cleaner to use that, instead of playing with _stdin_

Reply:
At first I went with `--password`, however, Docker complains that it's insecure to use it `--password` directly and suggests to use `--password-stdin` instead. See message below:

```WARNING! Using --password via the CLI is insecure. Use --password-stdin.```

In general, even if `password-stdin` is used, there's still another warning:
```
WARNING! Your password will be stored unencrypted in /home/pgrzesik/.docker/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store
```

But we cannot really do anything meaningful about that one if we want to automatically login people to `docker`.

Reply:
What do you think here? Should we go back to `--password` or keep the current approach (though I agree that it's so-so with `stdin` interaction)

Reply:
If it's just a printed warning when `--password` is used, and not a deprecation of any kind, I would switch to regular param input

Reply:
Changed :+1: